File: config/paths.py
"""Portable path helpers for thesis analysis scripts."""
from pathlib import Path
import os


def _root(env_var: str, default: str) -> Path:
    return Path(os.environ.get(env_var, default)).expanduser()


def analysis_path(*parts: str) -> Path:
    return _root(THESIS_ANALYSIS_DIR, data/analysis).joinpath(*parts)


def input_path(*parts: str) -> Path:
    return _root(THESIS_INPUT_DIR, data/input).joinpath(*parts)


def project_resource_path(*parts: str) -> Path:
    return _root(THESIS_RESOURCE_DIR, data/external).joinpath(*parts)


def external_path(*parts: str) -> Path:
    return _root(THESIS_EXTERNAL_ROOT, data/external).joinpath(*parts)

File: config/paths.R
# Portable path helpers for thesis analysis scripts.
#
# Set these environment variables before running scripts, or edit the defaults
# to match your local data layout. The GitHub repository does not include raw
# sequencing files or generated analysis outputs.

analysis_root <- function() {
  Sys.getenv(THESIS_ANALYSIS_DIR, unset = file.path(data, analysis))
}

input_root <- function() {
  Sys.getenv(THESIS_INPUT_DIR, unset = file.path(data, input))
}

project_resource_root <- function() {
  Sys.getenv(THESIS_RESOURCE_DIR, unset = file.path(data, external))
}

external_root <- function() {
  Sys.getenv(THESIS_EXTERNAL_ROOT, unset = file.path(data, external))
}

analysis_path <- function(...) file.path(analysis_root(), ...)
input_path <- function(...) file.path(input_root(), ...)
project_resource_path <- function(...) file.path(project_resource_root(), ...)
external_path <- function(...) file.path(external_root(), ...)
